akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -0,0 +1,403 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.Behavior.StoppedBehavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, MutableStashBuffer }
+import akka.event.Logging
+import akka.persistence.Eventsourced.{ PendingHandlerInvocation, StashingHandlerInvocation }
+import akka.persistence.JournalProtocol._
+import akka.persistence._
+import akka.persistence.journal.Tagged
+import akka.persistence.typed.scaladsl.PersistentBehaviors._
+import akka.{ actor ⇒ a }
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+
+abstract class EventsourcedRunning[Command, Event, State](
+  val context:                 ActorContext[Any],
+  val internalStash:           MutableStashBuffer[Any],
+  private var _lastSequenceNr: Long,
+  instanceId:                  Int, writerUuid: String // TODO WRAP THOSE INTO ONE 'identity'
+) extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement { same ⇒
```

Comment:
Could we perhaps just use "this" instead of giving it an alias, given that Behavior.same exists and is something else but very close to the same thing?

Reply:
since we have nested classes, `this` gets shadowed and is a pain to use in those classes, which is why that "same". I can move to using Behavior.same I think / hope in most places, I still need the alias though I believe, due to inner classes.

Reply:
Ah, I see